mountpoint-s3-client/src/mock_client.rs

```diff
@@ -120,6 +128,7 @@ pub struct MockClient {
     objects: Arc<RwLock<BTreeMap<String, MockObject>>>,
     in_progress_uploads: Arc<RwLock<BTreeSet<String>>>,
     operation_counts: Arc<RwLock<HashMap<Operation, u64>>>,
+    read_window_increment_failed: Arc<AtomicBool>,
```

Comment:
Why do we need a shared flag on the client, rather than independent ones on the response (and backpressure handle)?

Reply:
That is unfortunate, yes. This "invalidates" the whole client after a failed check. If we "invalidate" a single request, the errors are just not observable in tests. My suspicion is, that the failure most often occurs in `readahead` reads and kernel just ignores errors on those.